internal/pkg/service/buffer/storage/statistics/repository/repository.go

```diff
@@ -1,44 +1,74 @@
+// Package repository provides schema and database operations for storage statistics.
+//
+// # Data Model
+//
+// Statistics are stored in the etcd database as [statistics.Value] under the following key format:
+//
+//	storage/stats/<LEVEL:local>/<SLICE_KEY:PROJECT_ID/RECEIVER_ID/EXPORT_ID/FILE_ID/VOLUME_ID/SLICE_ID>/value
+//
+// Statistics are stored at the slice level, which represents the smallest unit.
+//
+// To obtain statistics for a parent object such as a file, export, receiver, or project,
+// simply sum up all the values under the corresponding prefix.
+//
+//	For a project:   storage/stats/<LEVEL>/<PROJECT_ID>/<RECEIVER_ID>/
+//	For a receiver:  storage/stats/<LEVEL>/<PROJECT_ID>/<RECEIVER_ID>/
+//	For an export:   storage/stats/<LEVEL>/<PROJECT_ID>/<RECEIVER_ID>/<EXPORT_ID>/
+//	For a file:      storage/stats/<LEVEL>/<PROJECT_ID>/<RECEIVER_ID>/<EXPORT_ID>/<FILE_ID>/
+//	For a slice:     storage/stats/<LEVEL>/<PROJECT_ID>/<RECEIVER_ID>/<EXPORT_ID>/<FILE_ID>/<VOLUME_ID>/<SLICE_ID>
+//
+// # Rollup
+//
+// Over time, files and slices expire and are deleted during the Cleanup operation.
+// Their statistics in the "local" and "staging" levels are also deleted.
+// Statistics in the "target" level are summarized under the key:
+//
+//	stats/<LEVEL:target>/<EXPORT_KEY:PROJECT_ID/RECEIVER_ID/EXPORT_ID>/_sum
 package repository
 
 import (
-	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	etcd "go.etcd.io/etcd/client/v3"
+
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/serde"
 	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
-	etcd "go.etcd.io/etcd/client/v3"
 )
 
 const (
-	// putMaxStatsPerTxn defines maximum number of keys per transaction when updating database values
+	// putMaxStatsPerTxn defines maximum number of keys per transaction when updating database values.
 	putMaxStatsPerTxn = 100
 )
 
-type atomicProvider = AtomicProvider
+type _provider = Provider
 
+// Repository provides database operations for storage statistics records.
 type Repository struct {
-	*atomicProvider
+	_provider
```

Comment:
`_provider` potrebuje pre embedding, male pismeno aby to bolo private (nedalo sa zmenit zvonku),

`provider struct` a `Provider interface` uz mame pouzite.